aptos-move/block-executor/src/executor.rs

```diff
@@ -509,13 +517,13 @@ where
                         &scheduler,
                         base_view,
                         role,
+                        &index_mapping,
```

Comment:
So that extra instance is needed for [here](https://github.com/aptos-labs/aptos-core/blob/145bb56335dafd62d1538077a2254af8814b82f6/aptos-move/block-executor/src/executor.rs#LL382C8-L382C8) and [here](https://github.com/aptos-labs/aptos-core/blob/145bb56335dafd62d1538077a2254af8814b82f6/aptos-move/block-executor/src/executor.rs#LL387C11-L387C11), which i agree is a bit overkill.

i see there are 4 options.
1. (current) make 3 IndexMapping instances, 1 held by `Scheduler`, 1 held by `LastInputOutput`, 1 for the extra stuff.
  - Pros: best readability, smallest diff
  - Cons: computationally most expensive
2. make 2 IndexMapping instances, 1 held by `Scheduler`, 1 held by `LastInputOutput`. For the extra stuff, borrow the one held by `Scheduler` (by either marking the IndexMapping field `pub` or making a backdoor scheduler API).
  - Pros: good readability except the backdoor stuff is ugly.
  - Cons: still expensive
3. Make 1 IndexMapping instance, passing its reference for every `Scheduler`/`LastInputOuput` method.
  - Pros: minimized computational cost
  - Cons: changes almost every `Scheduler`/`LastInputOutput` API. largest diff.
4. Make 1 IndexMapping instance, put in `Arc`, let `Scheduler`/`LastInputOutput` hold an `Arc<IndexMapping>`.
  - Pros: best readability, smallest diff
  - Cons: none. Arc overhead mainly comes from clones, but we only clone twice. Read-only access should be thread-safe without lock by default.

Will switch to 4 as it seems the best.

Reply:
@sitalkedia ^

Reply:
You still don't need to pass the index mapping explicitly to `coordinator_commit_hook` function as you have access to the scheduler and the scheduler can expose the underlying index mapping.

Reply:
We designed `IndexMapping` to handle all index work, `Scheduler` and `LastInputOutput` being the two main consumers, each given the access to `IndexMapping`: an `Arc<IndexMapping>`.

Now we have identified a 3rd consumer: `coordinator_commit_hook()`, why not just give it the access to the official service provider, and why bother the other consumers `Scheduler` and `LastInputOutput` :)